### PR TITLE
Cart cleanup adapters

### PR DESCRIPTION
**Pull request by danielpoe**
Current problems:
- there is no working emailadapter - name is misleading
- the cart adpater has an inmemory storage - but the adapter itself is not "inmemory"

Therefore this PR cleans this up and prepares at the same time that an additional module can provide a email adpater

### CODE REVIEW

File: cart/Readme.md [hunk 1]
```diff
@@ -21,6 +21,10 @@ The cart module and its services are used by the checkout module.
 
 ### Configurations
 
+For all possible configurations you can check the `module.go` (CueConfig function)
+As always you can also dumo the current configuration with the "config" Flamingo command.
```
Comment (tessig) [inline cart/Readme.md:25]:
```suggestion
As always you can also dump the current configuration with the "config" Flamingo command.
```

File: cart/infrastructure/placeorder/logger.go [hunk 2]
```diff
@@ -0,0 +1,155 @@
...
+	var placedOrders placeorder.PlacedOrderInfos
+	placedOrders = append(placedOrders, placeorder.PlacedOrderInfo{
+		OrderNumber: cart.ID,
+	})
+
+	return placedOrders, nil
+}
+
+// PlaceCustomerCart places a customer cart as order email
+func (e *PlaceOrderLoggerAdapter) PlaceCustomerCart(ctx context.Context, auth authDomain.Auth, cart *cartDomain.Cart, payment *placeorder.Payment) (placeorder.PlacedOrderInfos, error) {
```
Comment (tessig) [inline cart/infrastructure/placeorder/logger.go:67]: we could try to reduce this code duplication

Comment (danielpoe) [reply cart/infrastructure/placeorder/logger.go:54]: good point

**Comment by danielpoe** [comment]
@tessig @carstendietrich any update on that PR?

File: cart/infrastructure/defaultCartBehaviour.go [hunk 3]
```diff
@@ -346,57 +345,57 @@ func (cob *InMemoryBehaviour) CleanDelivery(ctx context.Context, cart *domaincar
...
 	cart.Purchaser = purchaser
 
 	if additionalData != nil {
 		cart.AdditionalData.CustomAttributes = additionalData.CustomAttributes
 	}
 
-	err := cob.cartStorage.StoreCart(cart)
+	err := cob.cartStorage.StoreCart(ctx, cart)
 	if err != nil {
 		return nil, nil, errors.Wrap(err, "cart.infrastructure.InMemoryBehaviour: error on saving cart")
```
Comment (tessig) [inline cart/infrastructure/defaultCartBehaviour.go:366]:
```suggestion
		return nil, nil, errors.Wrap(err, "cart.infrastructure.DefaultCartBehaviour: error on saving cart")
```

File: cart/infrastructure/defaultCustomerCartserviceAdapter.go [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+	}
+	return fmt.Sprintf("%v-%v", auth.IDToken.Subject, cartID), nil
+}
+
+// GetModifyBehaviour gets the cart order behaviour of the service
+func (cs *DefaultCustomerCartService) GetModifyBehaviour(context.Context, domain.Auth) (cart.ModifyBehaviour, error) {
+	return cs.defaultBehaviour, nil
+}
+
+// RestoreCart restores a previously used cart
```
Comment (tessig) [inline cart/infrastructure/defaultCustomerCartserviceAdapter.go:68]: perhaps directly mark as deprecated?

File: cart/infrastructure/defaultGuestCartserviceAdapter.go [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+func (gcs *DefaultGuestCartService) GetNewCart(ctx context.Context) (*cart.Cart, error) {
+	return gcs.defaultBehaviour.StoreNewCart(ctx, &cart.Cart{ID: strconv.Itoa(rand.Int())})
+}
+
+// GetModifyBehaviour returns the cart order behaviour of the service
+func (gcs *DefaultGuestCartService) GetModifyBehaviour(context.Context) (cart.ModifyBehaviour, error) {
+	return gcs.defaultBehaviour, nil
+}
+
+// RestoreCart restores a previously used guest cart
```
Comment (tessig) [inline cart/infrastructure/defaultGuestCartserviceAdapter.go:48]: perhaps directly mark as deprecated?

**Review by tessig — APPROVED** [review]

**Comment by tessig** [comment]
nice work, I suggest we do a squash here.

**Comment by danielpoe** [comment]
Closing in favor of new PR

**Pull request closed without merging**